src/CosmosEditorManager.ts

```diff
@@ -110,13 +110,14 @@ export class CosmosEditorManager {
         return editorFilePath;
     }
 
-    public async onDidSaveTextDocument(trackTelemetry: () => void, globalState: vscode.Memento, doc: vscode.TextDocument, tree: AzureTreeDataProvider): Promise<void> {
+    public async onDidSaveTextDocument(context: IActionContext, globalState: vscode.Memento, doc: vscode.TextDocument, tree: AzureTreeDataProvider): Promise<void> {
+        context.sendTelemetry = false;
         let filePath = Object.keys(this.fileMap).find((filePath) => path.relative(doc.uri.fsPath, filePath) === '');
         if (!filePath) {
             filePath = await this.loadPersistedEditor(doc.uri, tree);
         }
         if (!this.ignoreSave && filePath) {
-            trackTelemetry();
+            context.sendTelemetry = true;
```

Comment:
If I understand this correctly, this would send telemetry only on files which we handle. 

@StephenWeatherford is there a similar aim in the pipeline for the arm extension?

Reply:
I think Jay already fixed that, but I'll have to check.  Eric's changes are just a refactoring, shouldn't modify current behavior.